File: src/features/pix/redux/models/request/generateDynamicPixQrCodeRequest.ts
import { ApiRequest } from '_config'
import { ChangeType } from '../response/enum'
import { PixAddress } from '../response/PixAddress'

export interface GenerateDynamicPixQRCodeRequest extends ApiRequest {
  pixKey: string
  taxId: string
  payerTaxId: string
  payerName: string
  value: number
  address: PixAddress
  changeType: ChangeType
}

File: osb-core-main/Platform/Integration/Osb.Core.Platform.Integration.Entity/FitBank/Models/Request/GenerateDynamicPixQrCodeRequest.cs
using System;
using Osb.Core.Platform.Common.Entity.Enums;
using Osb.Core.Platform.Integration.Entity.Models.Request.Base;

namespace Osb.Core.Platform.Integration.Entity.FitBank.Models.Request
{
    public class GenerateDynamicPixQrCodeRequest : BaseRequest
    {
        public new string Method { get => "GenerateDynamicPixQRCode"; }
        public string PixKey { get; set; }
        public string TaxId { get; set; }
        public string PayerTaxId { get; set; }
        public string PayerName { get; set; }
        public decimal Value { get; set; }
        public DateTime ExpirationDate { get; set; }
        public string Bank { get; set; }
        public string BankBranch { get; set; }
        public string BankAccount { get; set; }
        public string BankAccountDigit { get; set; }
        public PixAddress Address { get; set; }
        public ChangeType ChangeType { get; set; }
    }
}